dbms/src/Functions/IFunction.cpp

```diff
@@ -656,11 +656,14 @@ DataTypePtr FunctionBuilderImpl::getReturnType(const ColumnsWithTypeAndName & ar
             arg.type = recursiveRemoveLowCardinality(arg.type);
         }
 
+        auto type_without_low_cardinality = getReturnTypeWithoutLowCardinality(args_without_low_cardinality);
+
         if (canBeExecutedOnLowCardinalityDictionary() && has_low_cardinality
-            && num_full_low_cardinality_columns <= 1 && num_full_ordinary_columns == 0)
-            return std::make_shared<DataTypeLowCardinality>(getReturnTypeWithoutLowCardinality(args_without_low_cardinality));
+            && num_full_low_cardinality_columns <= 1 && num_full_ordinary_columns == 0
```

Comment:
If `num_full_low_cardinality_columns` equals to 0, is it suitable to convert result type to LowCardinality?